tests/std/tests/P0896R4_ranges_alg_equal/test.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <concepts>
+#include <cstdlib>
+#include <ranges>
+//
+#include <range_algorithm_support.hpp>
+
+constexpr void smoke_test() {
+    using ranges::equal, ranges::equal_to;
+    using std::abort, std::array, std::pair, std::same_as;
+
+    array<pair<int, int>, 3> const x   = {{{0, 42}, {2, 42}, {4, 42}}};
+    array<pair<long, long>, 3> const y = {{{13, 0}, {13, 2}, {13, 4}}};
+
+    {
+        // Validate sized ranges
+        auto result = equal(x, y, equal_to{}, get_first, get_second);
+        STATIC_ASSERT(same_as<decltype(result), bool>);
+        assert(result);
+        assert(!equal(x, y, equal_to{}, get_first, get_first));
+    }
+    {
+        // Validate non-sized ranges
+        auto result = equal(move_only_range{x}, move_only_range{y}, equal_to{}, get_first, get_second);

```

Comment:
It seems like this should use something other than equal_to{}; otherwise we might accidentally depend on that. Perhaps equal_to_second_plus_1 ? (That would also check for transposed parameters)